### PR TITLE
Add retry mechanism for streaming api connection while consuming events.

### PR DESCRIPTION
**Pull request by dcojo**

### CODE REVIEW

File: src/main/java/streamingapi/client/StreamMonitor.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package streamingapi.client;
+
+/**
+ * Monitor used stop the execution or to check if the execution should continue.
```
Comment (ana-puiu) [inline src/main/java/streamingapi/client/StreamMonitor.java]: typo: used *to* stop

File: src/main/java/streamingapi/client/StreamMonitor.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * @author dancojocaru
+ */
+public class StreamMonitor {
+
+	private boolean isActive = true;
+
+	/**
+	 * Stops the monitored execution
+	 * {@code false} represents
```
Comment (ana-puiu) [inline src/main/java/streamingapi/client/StreamMonitor.java]: represents .. ? i think this is incomplete

File: src/main/java/streamingapi/client/StreamingApiClient.java [hunk 3]
```diff
@@ -50,32 +51,42 @@
...
-	String CONTENT_TYPE_VALUE = "application/json";
+	private final static Logger LOGGER = Logger.getLogger(StreamingApiClient.class.getName());
 
-	String API_KEY = "apikey";
-	String MAMBU_STREAM_ID_HEADER_FIELD = "X-Mambu-StreamId";
+	private static final int TIMEOUT_UNTIL_RETRY_CONNECTION_IN_MILLIS = 2000;
+	private static final String LOST_CONNECTION_TO_STREAMING_API_MESSAGE = "Lost connection to streaming api.";
+	private static final String TRYING_TO_REINITIALIZE_CONNECTION_TO_STREAMING_API_MESSAGE = "Trying to reinitialize connection to streaming api.";
+
+	private String MAMBU_ENDPOINT = "http://demo_tenant.localhost:8000";
```
Comment (ana-puiu) [inline src/main/java/streamingapi/client/StreamingApiClient.java]: some of these strings can be static final, since they are constants

File: src/main/java/streamingapi/client/StreamMonitor.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+	 */
+	public void stopExecution() {
```
Comment (ana-puiu) [inline src/main/java/streamingapi/client/StreamMonitor.java]: This is never used. Is this intended ?

File: src/main/java/streamingapi/client/StreamingApiClient.java [hunk 5]
```diff
@@ -97,33 +108,43 @@ public Subscription createSubscription(Subscription subscription, String streami
 
 	/**
 	 * Consume events of based on a subscription.
+	 * When the connection to streaming api is lost, the connection is reinitialized if the monitor approves.
```
Comment (georgeghimici) [inline src/main/java/streamingapi/client/StreamingApiClient.java]: I would rephrase the Info to: 
The connection to streaming api can be lost when the streaming api events store closes the stream because the stream timeout is reached or due to unpredictable network events.

File: src/main/java/streamingapi/client/StreamingApiClient.java [hunk 6]
```diff
@@ -97,33 +108,43 @@ public Subscription createSubscription(Subscription subscription, String streami
...
 
-					processor.process(batch.getEvents());
-					Response response = commitCursor(batch.getCursor(), streamId, subscriptionId, streamingApiKey);
-					if (!isCommitCursorSuccessful(response)) {
-						throw new CommitCursorException("Error while committing cursor.");
+							processor.process(batch.getEvents());
+							Response response = commitCursor(batch.getCursor(), streamId, subscriptionId, streamingApiKey);
+							if (!isCommitCursorSuccessful(response)) {
+								throw new CommitCursorException("Error while committing cursor.");
+							}
```
Comment (georgeghimici) [inline src/main/java/streamingapi/client/StreamingApiClient.java]: I think the out. println can be removed

File: src/main/java/streamingapi/client/StreamMonitor.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
+package streamingapi.client;
+
+/**
+ * Monitor used stop the execution or to check if the execution should continue.
+ *
+ * @author dancojocaru
+ */
+public class StreamMonitor {
+
```
Comment (georgeghimici) [inline src/main/java/streamingapi/client/StreamMonitor.java]: I see that this value is not customised from the tests, it isn't  necessary ?

File: src/main/java/streamingapi/client/StreamMonitor.java (hunk 4 above)
Comment (dcojo) [reply src/main/java/streamingapi/client/StreamMonitor.java]: Right now, yes. I added it just as an example. To use it, a new async execution of the StreamingApiClient should be started, and then the main thread could stop it.

File: src/main/java/streamingapi/client/StreamMonitor.java (hunk 7 above)
Comment (dcojo) [reply src/main/java/streamingapi/client/StreamMonitor.java]: The default value is true, so it wont require initialisation. I'm not sure what do you mean by "customised".

Comment (georgeghimici) [reply src/main/java/streamingapi/client/StreamMonitor.java]: Got it. Customised in the sense that is not used from the test with a different value
